From d5a1f365bfd4420912735709bcb8004d3ad7226c Mon Sep 17 00:00:00 2001
From: Joel A Fernandes <agnel.joel@gmail.com>
Date: Fri, 29 Jul 2011 17:23:43 -0500
Subject: [PATCH] v4l2src: Add UYVY color format to the list of enumerations

OMAP3 ISP returns -EINVAL for VIDIOC_ENUM_FMT ioctl on video node. Due to this, format negotiation fails. This adds UYVY format to the list of enumerations.

Signed-off-by: Joel A Fernandes <agnel.joel@gmail.com>
---
 sys/v4l2/gstv4l2object.c |   17 +++++++++++++++++
 1 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 7e15489..e04c0fd 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1025,6 +1025,23 @@ gst_v4l2_object_fill_format_list (GstV4l2Object * v4l2object)
         (GCompareFunc) format_cmp_func);
   }
 
+/* Add UYVY format to the list of enumerations */
+  format = g_new0 (struct v4l2_fmtdesc, 1);
+  format->type = (enum v4l2_buf_type)-1;
+  format->index = n;
+  format->flags = 0;
+  format->pixelformat = V4L2_PIX_FMT_UYVY;
+  v4l2object->formats = g_slist_insert_sorted (v4l2object->formats, format,
+      (GCompareFunc) format_cmp_func);
+
+  GST_INFO_OBJECT (v4l2object->element, "index:       %u", format->index);
+  GST_INFO_OBJECT (v4l2object->element, "type:        %d", format->type);
+  GST_INFO_OBJECT (v4l2object->element, "flags:       %08x", format->flags);
+  GST_INFO_OBJECT (v4l2object->element, "description: '%s'",
+      format->description);
+  GST_INFO_OBJECT (v4l2object->element, "pixelformat: %" GST_FOURCC_FORMAT,
+        GST_FOURCC_ARGS (format->pixelformat));
+
 #ifndef GST_DISABLE_GST_DEBUG
   {
     GSList *l;
-- 
1.7.0.4

